ompi/mpi/c/session_get_nth_pset.c

```diff
@@ -31,12 +31,14 @@ int MPI_Session_get_nth_pset (MPI_Session session, MPI_Info info, int n, int *le
     int rc = MPI_SUCCESS;
 
     if ( MPI_PARAM_CHECK ) {
-        if (NULL == session || (NULL == pset_name && *len > 0) || n < 0) {
+        if (ompi_instance_invalid(session)) {
             if (NULL != session) {
-                return OMPI_ERRHANDLER_INVOKE(session, MPI_ERR_ARG, FUNC_NAME);
+                return OMPI_ERRHANDLER_INVOKE(session, MPI_ERR_SESSION, FUNC_NAME);
             } else {
-                return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_ARG, FUNC_NAME);
+                return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_SESSION, FUNC_NAME);
             }
+        } else if ((NULL == pset_name && *len > 0) || n < 0) {
```

Comment:
@hppritcha If `n` is larger or equal than the number of psets, it should also be an error. Would it be too expensive to call `ompi_instance_get_num_psets()` to check that `n` is in the range `[0, num_psets)` ?

Reply:
i don't think this would be too expensive.  this is supposedly a relatively infrequently called method, so adding a sanity check, esp. with --enable-debug set when configuring ompi, would be problematic.   i'll open a pr to add this check.